Content.Server/Implants/RadioImplantSystem.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Content.Server.Radio.Components;
+using Content.Shared.FixedPoint;
+using Content.Shared.Implants;
+using Content.Shared.Implants.Components;
+using Content.Shared.Tag;
+using Robust.Server.GameObjects;
+using Robust.Shared.Containers;
+
+namespace Content.Server.Implants;
+
+public sealed class RadioImplantSystem : EntitySystem
+{
+    [Dependency] private readonly TagSystem _tag = default!;
+    [Dependency] private readonly IServerEntityManager _entityManager = default!;
+
+
+    [ValidatePrototypeId<TagPrototype>]
+    public const string RadioImplantTag = "RadioImplant";
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RadioImplantComponent, ImplantImplantedEvent>(RadioImplantCheck);
+        SubscribeLocalEvent<RadioImplantComponent, EntGotRemovedFromContainerMessage>(OnRemove);
+    }
+
+    /// <summary>
+    /// Checks if the implant was a Radio Implant or not
+    /// </summary>
+    public void RadioImplantCheck(EntityUid uid, RadioImplantComponent comp, ref ImplantImplantedEvent ev)
+    {
+        if (!_tag.HasTag(ev.Implant, RadioImplantTag) || ev.Implanted == null)
```

Comment:
Did some testing and I don't think the tag check is necessary! If an implanter has the `RadioImplantComponent` it should always give the radio channels. The tag isn't necessary I don't think 🤔